src/hotspot/share/runtime/arguments.cpp

```diff
@@ -2267,7 +2267,11 @@ jint Arguments::parse_each_vm_init_arg(const JavaVMInitArgs* args, bool* patch_m
     } else if (match_option(option, "--patch-module=", &tail)) {
       // --patch-module=<module>=<file>(<pathsep><file>)*
       int res = process_patch_mod_option(tail, patch_mod_javabase);
-      if (res != JNI_OK) {
+      if (res == JNI_OK) {
+        // Add jdk.patched system property when processing of args was OK
+        PropertyList_unique_add(&_system_properties, "jdk.patched", "true",
```

Comment:
Hello Severin, in addition should we explicitly set this system property to `false` (or remove the system property) when `--patch-module` isn't used? I think that would then prevent applications from passing `-Djdk.patched=true` when the runtime hasn't been patched. 
While at it, should there be additional guards to prevent applications from changing the value of this system property by doing `System.setProperty("jdk.patched")` at runtime?

Reply:
> in addition, should we explicitly set this system property to false (or remove the system property) when --patch-module isn't used? I think that would then prevent applications from passing -Djdk.patched=true when the runtime hasn't been patched.

Sure, I can add this.

> While at it, should there be additional guards to prevent applications from changing the value of this system property by doing System.setProperty("jdk.patched") at runtime?

The JVM sets this property as read-only. So setting `jdk.patched=true` at command line or in code won't work to override. I'll add tests for these scenarios.

Reply:
> While at it, should there be additional guards to prevent applications from changing the value of this system property by doing `System.setProperty("jdk.patched")` at runtime?

The updated patch has a test for this. In a nutshell any JDK property can be changed in code (also `jdk.module.patch.0` if desired). So this would be no different. The javadoc warns about unintended consequences of setting JDK properties that way.